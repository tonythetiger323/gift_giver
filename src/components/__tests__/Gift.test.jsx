import React from "react";
import {shallow} from "enzyme";
import Gift from "../Gift";

describe("Gift", () => {
  const mockRemove = jest.fn();
  const id = 1;
  const props = {gift: {id}, removeGift: mockRemove};
  const gift = shallow(<Gift {...props} />);

  it("renders properly", () => {
    expect(gift).toMatchSnapshot();
  });

  it("initializes a person and present in `state`", () => {
    expect(gift.state()).toEqual({person: "", present: ""});
  });

  describe("when typing into the person input", () => {
    const person = "Daddy";

    beforeEach(() => {
      gift.find(".input-person").simulate("change", {target: {value: person}});
    });

    it("updates the person in `state`", () => {
      expect(gift.state().person).toEqual(person);
    });
  });

  describe("when typing into the present input", () => {
    const present = "Box of Wine";

    beforeEach(() => {
      gift.find(".input-present").simulate("change", {target: {value: present}});
    });

    it("updates the present in `state`", () => {
      expect(gift.state().present).toEqual(present);
    });
  });
});
