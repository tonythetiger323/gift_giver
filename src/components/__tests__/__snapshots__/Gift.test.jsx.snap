// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift renders properly 1`] = `
<div
  className="gift"
>
  <Form
    bsClass="form"
    componentClass="form"
    horizontal={false}
    inline={false}
  >
    <FormGroup
      bsClass="form-group"
    >
      <ControlLabel
        bsClass="control-label"
        srOnly={false}
      >
        Person
      </ControlLabel>
      <FormControl
        bsClass="form-control"
        className="input-person"
        componentClass="input"
        onChange={[Function]}
      />
    </FormGroup>
    <FormGroup
      bsClass="form-group"
    >
      <ControlLabel
        bsClass="control-label"
        srOnly={false}
      >
        Present
      </ControlLabel>
      <FormControl
        bsClass="form-control"
        className="input-present"
        componentClass="input"
        onChange={[Function]}
      />
    </FormGroup>
  </Form>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    className="btn-remove"
    disabled={false}
    onClick={[Function]}
  >
    Remove Gift
  </Button>
</div>
`;
